src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -254,4 +297,111 @@ impl Coordinator {
             ObjectType::Cluster,
         )))
     }
+
+    pub(crate) async fn finalize_alter_cluster_managed_to_managed(
+        &mut self,
+        session: &Session,
+        AlterClusterPlan {
+            id: cluster_id,
+            name: cluster_name,
+            ..
+        }: AlterClusterPlan,
+        new_config: ClusterVariantManaged,
+    ) -> Result<StageResult<Box<ClusterStage>>, AdapterError> {
+        let cluster = self.catalog.get_cluster(cluster_id);
+        let mut ops = vec![];
+
+        // Gather the ops to remove the non pending replicas
+        // Also skip any billed_as free replicas
+        let remove_replicas = cluster
+            .replicas()
+            .filter_map(|r| {
+                if !r.config.location.pending() && !r.config.location.internal() {
+                    Some(catalog::DropObjectInfo::ClusterReplica((
+                        cluster_id.clone(),
+                        r.replica_id,
+                        ReplicaCreateDropReason::Manual,
+                    )))
+                } else {
+                    None
+                }
+            })
+            .collect();
+        ops.push(catalog::Op::DropObjects(remove_replicas));
+
+        // Gather the Ops to remove the "-pending" suffix from the name and set
+        // pending to false
+        let finalize_replicas: Vec<catalog::Op> = cluster
+            .replicas()
+            .filter_map(|r| {
+                if r.config.location.pending() {
+                    let cluster_ident = match Ident::new(cluster.name.clone()) {
+                        Ok(id) => id,
+                        Err(err) => return Some(Err(err)),
```

Comment:
I would wrap this error and the one below in some form of internal error. Erroring here would be unexpected and a result of programmer error and I think these Errors (`IdentError`) would probably be confusing to the user.

Reply:
Ok, I wrapped these `AdapterError::Internal`, I don't think it should be possible to hit this error.